cli/pkg/kctrl/cmd/package/release/helm_openapi_schema_gen.go

```diff
@@ -0,0 +1,362 @@
+					v.Content[0].HeadComment = v.HeadComment
+				}
+				switch v.Kind {
+				case yaml3.MappingNode:
+					calculatedProperties, err := h.calculateProperties(nil, v)
+					if err != nil {
+						return nil, err
+					}
+					for _, item := range calculatedProperties.Items {
+						if item.Key == propertiesKey {
+							properties.Items = append(properties.Items, item.Value.(*Map).Items...)
+						}
+					}
+				case yaml3.SequenceNode:
+					calculatedProperties, err := h.calculateProperties(nil, v)
+					if err != nil {
+						return nil, err
+					}
+					for _, item := range calculatedProperties.Items {
+						if item.Key == itemsKey {
+							properties.Items = append(properties.Items, item.Value.(*Map).Items...)
+						}
+					}
+				case yaml3.ScalarNode:
+					val, err := h.getDefaultValue(h.openAPITypeFor(value.Content[0].Tag, value.Content[0].Value), v.Value)
+					if err != nil {
+						return nil, err
+					}
+					defaultVals = append(defaultVals, val)
+				case yaml3.AliasNode:
+					calculatedProperties, err := h.calculateProperties(nil, v.Alias)
+					if err != nil {
+						return nil, err
+					}
+					for _, item := range calculatedProperties.Items {
+						if item.Key == itemsKey {
+							properties.Items = append(properties.Items, item.Value.(*Map).Items...)
+						}
+					}
+				default:
+					return nil, fmt.Errorf("Unrecognized type %T", v.Kind)
+				}
+			}
+
+			var itemsProperties *Map
+			switch value.Content[0].Kind {
+			case yaml3.MappingNode, yaml3.AliasNode:
+				itemsProperties = &Map{[]*MapItem{
+					&MapItem{Key: typeKey, Value: "object"},
+					&MapItem{Key: propertiesKey, Value: properties}}}
+			case yaml3.SequenceNode:
+				itemsProperties = &Map{[]*MapItem{
+					&MapItem{Key: typeKey, Value: "array"},
+					&MapItem{Key: defaultKey, Value: "[]"},
+					&MapItem{Key: itemsKey, Value: properties}}}
+			case yaml3.ScalarNode:
+				itemsProperties = &Map{[]*MapItem{
+					&MapItem{Key: typeKey, Value: h.openAPITypeFor(value.Content[0].Tag, value.Content[0].Value)}}}
+			}
+			apiKeys = append(apiKeys, &MapItem{Key: itemsKey, Value: itemsProperties})
+		}
+		apiKeys = append(apiKeys, &MapItem{Key: defaultKey, Value: defaultVals})
+		sort.Sort(apiKeys)
+		if key == nil {
+			return &Map{Items: apiKeys}, nil
+		}
+		return &Map{Items: []*MapItem{&MapItem{Key: key.Value, Value: &Map{Items: apiKeys}}}}, nil
+	case yaml3.ScalarNode:
+		var apiKeys openAPIKeys
+		description, isPresent := h.getDescriptionFromNode(key)
+		if isPresent {
+			apiKeys = append(apiKeys, &MapItem{Key: descriptionKey, Value: description})
+		}
+
+		valType := h.openAPITypeFor(value.Tag, value.Value)
+		defaultVal, err := h.getDefaultValue(valType, value.Value)
+		if err != nil {
+			return nil, err
+		}
+		apiKeys = append(apiKeys, &MapItem{Key: typeKey, Value: valType})
+		apiKeys = append(apiKeys, &MapItem{Key: defaultKey, Value: defaultVal})
+		if value.Tag == floatTag {
```

Comment:
Seems a bit weird to just check for the floatTag 🤔

Reply:
in openapi Spec, for a float value, type is number but format is float. Even ytt does the same.